apps/web/src/components/data-table/data-table-pagination.tsx

```diff
@@ -15,78 +17,102 @@ import {
   SelectValue,
 } from "@openstatus/ui";
 
-interface DataTablePaginationProps<TData> {
-  table: Table<TData>;
+import useUpdateSearchParams from "@/hooks/use-update-search-params";
+
+interface DataTablePaginationProps {
+  hasNextPage: boolean;
 }
 
-export function DataTablePagination<TData>({
-  table,
-}: DataTablePaginationProps<TData>) {
+export function DataTablePagination({ hasNextPage }: DataTablePaginationProps) {
+  const searchParams = useSearchParams();
+  const offset = parseInt(useSearchParams().get("offset") || "0");
+
+  const router = useRouter();
+  const pathname = usePathname();
+  const updateSearchParams = useUpdateSearchParams();
+
+  const updateSearchParamsWithRouter = (
+    query: Parameters<typeof updateSearchParams>[0],
+  ) => {
+    const updatedSearchParams = updateSearchParams(query);
+    return () => {
+      router.push(`${pathname}/?${updatedSearchParams.toString()}`);
```

Comment:
I don't think you need the `pathname` here. Also, a `.replace` might be better. Otherwise it will stack a new route on top of the history.

```suggestion
      router.replace(`?${updatedSearchParams.toString()}`);
```

Reply:
@mxkaske Thanks!. I will look into that.

Reply:
@mxkaske Changes made. Works as expected